### DISCUSSION TITLE
correct way to transclude in js widget

### DISCUSSION THREAD

**Discussion by welford** [TiddlyWiki/TiddlyWiki5#5690] [Q&A] [answered]
Hello,

I've written a little plugin in TW that transcludes a tiddler, but i'm struggling to do this correctly whent he trancluded widget itself uses a subsequently transcluded tiddler.

in the attached example. "study fruits" is trying to programatically include "banana" in this case, but it is not quite right. 

![image](https://user-images.githubusercontent.com/4590399/118459705-e928b780-b6f3-11eb-81cc-62ae6a99a190.png)

the code is:

```
var ShowCard = function (tiddler) {
   var tid = $tw.wiki.getTiddler(tiddler);
   //console.log(tiddler);
   var t = $tw.wiki.makeTranscludeWidget(tiddler, {document:document, variables:tid.fields});
   card.innerHTML = "";
   t.render(card,null);
   activeCard = tiddler;
}
```

Card is a dom node div i have created to be rendered in to, and i pass in "banana" as tiddler in this case.  Could someone point me in the right direction for achieving transclusion e.g. {{banana}} within a js widget.

[broken_transclude.zip](https://github.com/Jermolene/TiddlyWiki5/files/6492441/broken_transclude.zip)

**Comment by saqimtiaz** [comment, marked as answer]
```
   var t = $tw.wiki.makeTranscludeWidget(tiddler, {document:document, variables:{"currentTiddler":tiddler}});
}
```

**Reply by welford** [reply]
thanks!
